Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -778,6 +796,63 @@ void Device::blit_to(Gfx::Bitmap& target)
 
     Gfx::Painter painter { target };
     painter.blit({ 0, 0 }, *m_render_target, m_render_target->rect(), 1.0f, false);
+
+    if (ENABLE_STATISTICS_OVERLAY)
+        draw_statistics_overlay(target);
+}
+
+void Device::draw_statistics_overlay(Gfx::Bitmap& target)
+{
+    static Core::ElapsedTimer timer;
+    static String debug_string;
+    static int frame_counter;
+
+    frame_counter++;
+    int milliseconds = 0;
+    if (timer.is_valid())
+        milliseconds = timer.elapsed();
+    else
+        timer.start();
+
+    Gfx::Painter painter { target };
+
+    if (milliseconds > 500) {
+
+        if (g_num_pixels == 0)
+            g_num_pixels = 1;
+
+        int num_rendertarget_pixels = m_render_target->width() * m_render_target->height();
+
+        StringBuilder builder;
+        builder.append(String::formatted("Timings      : {}ms {}FPS\n", milliseconds / frame_counter, (milliseconds > 0) ? 1000 * frame_counter / milliseconds : 9999));
```

Comment:
This FPS calculation is an `int`, would be nice to get sub-frame precision (one decimal?) as long as we're not hitting our 144Hz target yet.